### PR TITLE
Minor changes to ROCm PyTorch

### PR DESCRIPTION
**Pull request by Jorghi12**
- Temporary solution for cudaOccupancyMaxActiveBlocksPerMultiprocessor until https://github.com/ROCm-Developer-Tools/HIP/issues/567 is handled.
- Removed unnecessary file SoftMaxCommon.cuh from disabled_features.yaml
- Enabled HCC with ThinLTO Linking

### CODE REVIEW

**Comment by Jorghi12** [comment]
LLVM OOM. Rebuilding rocmnightly here: https://ci.pytorch.org/jenkins/job/pytorch-builds/job/py2-clang3.8-rocmnightly-ubuntu16.04-build/2838/console

File: .jenkins/pytorch/build.sh [hunk 1]
```diff
@@ -30,10 +30,16 @@ cmake --version
 pip install -r requirements.txt || true
 
 if [[ "$BUILD_ENVIRONMENT" == *rocm* ]]; then
+  # This is necessary in order to cross compile (or else we'll have missing GPU device).
```
Comment (iotamudelta) [inline .jenkins/pytorch/build.sh]: one can target multiple GPU targets. we should consider doing so at least for fiji (gfx803).
see: https://github.com/RadeonOpenCompute/hcc#use-hcc_amdgpu_target-env-var

File: tools/amd_build/disabled_features.yaml [hunk 2]
```diff
@@ -23,16 +23,10 @@
...
-        "path": "aten/src/THCUNN/SoftMaxCommon.cuh",
-        "functions": {
-          "cudaOccupancyMaxActiveBlocksPerMultiprocessor": "16"
-        }
-      },
       {
         "path": "aten/src/ATen/native/cuda/SoftMax.cu",
         "functions": {
-          "cudaOccupancyMaxActiveBlocksPerMultiprocessor": "16"
+          "cudaOccupancyMaxActiveBlocksPerMultiprocessor": "max_active_blocks = 16;"
```
Comment (iotamudelta) [inline tools/amd_build/disabled_features.yaml]: I'm working on the disabled_features right now (see #30 ). I'm #ifdef ing this in the source. We can discuss what is preferable in #30. Please remove changes to disabled_features here.

File: cmake/public/LoadHIP.cmake [hunk 3]
```diff
@@ -102,12 +102,14 @@ FIND_PACKAGE(HIP 1.0)
 IF(HIP_FOUND)
   set(PYTORCH_FOUND_HIP TRUE)
 
+  ### Remove these when FindHIP.CMake PR is accepts these adjustments###
```
Comment (iotamudelta) [inline cmake/public/LoadHIP.cmake]: Reference PR link/number.

File: .jenkins/pytorch/build.sh [hunk 4]
```diff
@@ -30,10 +30,16 @@ cmake --version
...
   export HCC_AMDGPU_TARGET=gfx900
+
+  # These environment variables are necessary in order to avoid errors in HIP utility scripts.
   export LANG=C.UTF-8
   export LC_ALL=C.UTF-8
 
+  # This environment variable enabled HCC Optimizations that speed up the linking stage.
```
Comment (iotamudelta) [inline .jenkins/pytorch/build.sh]: reference https://github.com/RadeonOpenCompute/hcc#hcc-with-thinlto-linking

File: caffe2/CMakeLists.txt [hunk 5]
```diff
@@ -273,9 +273,8 @@ if(USE_ROCM)
   target_include_directories(caffe2_hip PRIVATE ${Caffe2_HIP_INCLUDES})
   target_include_directories(caffe2_hip INTERFACE $<INSTALL_INTERFACE:include>)
 
-  IF(BUILD_ATEN)
-    aten_set_target_props(caffe2_hip)
-  ENDIF()
+  # Set standard properties on the target
```
Comment (iotamudelta) [inline caffe2/CMakeLists.txt]: can this now be done unconditionally? @petrex please comment here

File: .jenkins/pytorch/build.sh [hunk 6]
```diff
@@ -30,10 +30,16 @@ cmake --version
 pip install -r requirements.txt || true
 
 if [[ "$BUILD_ENVIRONMENT" == *rocm* ]]; then
+  # This is necessary in order to cross compile (or else we'll have missing GPU device).
   export HCC_AMDGPU_TARGET=gfx900
+
+  # These environment variables are necessary in order to avoid errors in HIP utility scripts.
```
Comment (iotamudelta) [inline .jenkins/pytorch/build.sh]: unclear what the underlying issue is. is this a HIP shortcoming? what scripts? what errors?

File: caffe2/CMakeLists.txt (hunk 5 above)
Comment (Jorghi12) [reply caffe2/CMakeLists.txt]: @iotamudelta This should actually have always been done unconditionally. Notice that CUDA builds have it unconditional here: https://github.com/pytorch/pytorch/blob/master/caffe2/CMakeLists.txt#L249

File: .jenkins/pytorch/build.sh (hunk 6 above)
Comment (Jorghi12) [reply .jenkins/pytorch/build.sh]: These environment variables are not set on CI when we were running as the Jenkins user. The HIP Utility scripts require these environment variables to be set in order to run without error.

File: .jenkins/pytorch/build.sh (hunk 1 above)
Comment (Jorghi12) [reply .jenkins/pytorch/build.sh]: @iotamudelta The only issue is increasing build times & memory usage.

File: .jenkins/pytorch/build.sh (hunk 4 above)
Comment (Jorghi12) [reply .jenkins/pytorch/build.sh]: Added.

File: cmake/public/LoadHIP.cmake (hunk 3 above)
Comment (Jorghi12) [reply cmake/public/LoadHIP.cmake]: Added.

**Comment by Jorghi12** [comment]
Changes handled.

**Comment by iotamudelta** [comment]
It's missing the change for disabled_features. Otherwise it's ready to merge, I'd like to include it in our PR to upstream today.

**Comment by Jorghi12** [comment]
@iotamudelta

In order to not affect the if/def'd solution, the field should be completely removed from the disable_features.yaml.

**Comment by iotamudelta** [comment]
@pytorchbot retest this please

Assuming we can get all tests to pass, will merge after week28 PR w/ upstream is merged. Thanks!

**Comment by Jorghi12** [comment]
@pytorchbot retest this please

LLVM OOM :(

**Comment by Jorghi12** [comment]
Disabling KMTHINLTO. It's probably increasing the memory requirements.

**Pull request merged**